x-pack/plugins/ml/public/application/components/ml_inference/components/on_failure_configuration.tsx

```diff
@@ -0,0 +1,295 @@
+                        checked={actionOnFailure}
+                        onChange={(e: EuiSwitchEvent) => setActionOnFailure(e.target.checked)}
+                      />
+                    </EuiFormRow>
+                  </EuiFlexItem>
+                </EuiFlexGroup>
+              </EuiFlexItem>
+              <EuiFlexItem grow={false}>
+                <EuiFormRow
+                  fullWidth
+                  label={
+                    <EuiText size="s">
+                      <strong>
+                        {i18n.translate(
+                          'xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.advanced.onFailureHeadingLabel',
+                          { defaultMessage: 'Actions to take on failure' }
+                        )}
+                      </strong>
+                    </EuiText>
+                  }
+                  labelAppend={
+                    <EuiFlexGroup gutterSize="xs" justifyContent="flexEnd">
+                      <EuiFlexItem grow={false}>
+                        <EuiButtonEmpty
+                          iconType="pencil"
+                          size="xs"
+                          disabled={actionOnFailure === false}
+                          onClick={() => {
+                            setEditOnFailure(!editOnFailure);
+                          }}
+                        >
+                          {editOnFailure ? CANCEL_EDIT_MESSAGE : EDIT_MESSAGE}
+                        </EuiButtonEmpty>
+                      </EuiFlexItem>
+                      <EuiFlexItem grow={false}>
+                        {editOnFailure ? (
+                          <SaveChangesButton
+                            onClick={updateOnFailure}
+                            disabled={isOnFailureValid === false}
+                          />
+                        ) : null}
+                      </EuiFlexItem>
+                      <EuiFlexItem grow={false}>
+                        {editOnFailure ? (
+                          <EuiButtonEmpty size="xs" onClick={resetOnFailure}>
+                            {i18n.translate(
+                              'xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.advanced.resetOnFailureButton',
+                              { defaultMessage: 'Reset' }
+                            )}
+                          </EuiButtonEmpty>
+                        ) : null}
+                      </EuiFlexItem>
+                    </EuiFlexGroup>
+                  }
+                  helpText={
+                    actionOnFailure === true ? (
+                      <FormattedMessage
+                        id="xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.advanced.onFailureHelpText"
+                        defaultMessage="In case of failure, this configuration sets the index document as failed, provides the timestamp at which ingest failed, and the context for the failure."
```

Comment:
```suggestion
                        defaultMessage="In case of failure, this configuration stores the document in the dedicated `failed-<destination>` index, provides the timestamp at which ingest failed, and the context for the failure."
```